### PR TITLE
docs(github-app): document how to retrieve username and email of the …

### PR DESCRIPTION
**Pull request by m4s-b3n**
…github app service account

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,22 +35,46 @@ A GitHub app with the following permissions is required:
 - `Read access to metadata`
 - `Read and write access to administration, code, and issues`
 
-### Variables
+To be able to perform git operations using the apps identity retrieved the apps username and email as follows:
```
Comment (aatmmr) [inline README.md:38]:
```suggestion
```

File: README.md [hunk 2]
```diff
@@ -35,22 +35,46 @@ A GitHub app with the following permissions is required:
...
+
+It is possible to perform git operations using the apps identity. Retrieve the required username and email as follows:
```
Comment (aatmmr) [inline README.md:40]:
```suggestion
It is possible to perform git and GitHub operations using the apps identity. Retrieve the required username and email as follows:
```

File: README.md [hunk 3]
```diff
@@ -35,22 +35,46 @@ A GitHub app with the following permissions is required:
...
-| Key          | Description                         |
-| ------------ | ----------------------------------- |
-| GH_APP_ID    | ID of the [GitHub App](#github-app) |
-| ORGANIZATION | GitHub organization                 |
+| Key                 | Description                                |
+| ------------------- | ------------------------------------------ |
+| GH_APP_ID           | ID of the [GitHub App](#github-app)        |
+| ORGANIZATION        | GitHub organization                        |
+| CI_COMMIT_USER      | Name of the user for automated git tasks*  |
+| CI_COMMIT_USER_MAIL | Email of the user for automated git tasks* |
```
Comment (aatmmr) [inline README.md:75]:
```suggestion
| CI_COMMIT_USER      | Name of the user for automated git and GitHub tasks*  |
| CI_COMMIT_USER_MAIL | Email of the user for automated git and GitHub tasks* |
```

File: README.md [hunk 4]
```diff
@@ -35,22 +35,46 @@ A GitHub app with the following permissions is required:
...
+
+# retrieve app ID from github
+ID=$(curl -s -g https://api.github.com/users/$USER | jq '.id')
+
+# display data
+echo
+echo CI_COMMIT_USER=$USER
+echo CI_COMMIT_USER_MAIL=$ID+$USER@users.noreply.github.com
+```
+
```
Comment (aatmmr) [inline README.md:57]:
```suggestion
As an alternative, the user name and email can be fetched via browser as well. Open a browser and use the following url

```
https://api.github.com/users/{name-of-github-app}[bot]
```

where `{name-of-github-app}` is the name assigned to the app in the respective app settings. It is important that `[bot]` follows the name of the app.

The request returns a JSON that is displayed in the browser. From that request the value of the key `"id"` is required to assemble the user name and email address as following

*User Name*

```
{name-of-github-app}[bot]
```

*User Email*

```
{user-id}+{name-of-github-app}[bot]@users.noreply.github.com
```

where `{user-id}` is the value of the response key `"id"` and `{name-of-github-app}` is the name of the app. Remember to add `[bot]`.
```

Add browser way additionally?

**Review by aatmmr — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (m4s-b3n) [reply README.md:75]: only for git tasks as this is configured as git user

File: README.md (hunk 2 above)
Comment (m4s-b3n) [reply README.md:40]: only used for git

File: README.md (hunk 4 above)
Comment (m4s-b3n) [reply README.md:57]: suggestion looks broke 😮

Comment (aatmmr) [reply README.md:56]: 🔧 Fixed @m4s-b3n 🔧

File: README.md (hunk 2 above)
Comment (aatmmr) [reply README.md:39]: Removed the GitHub part from the suggestion @m4s-b3n

File: README.md (hunk 4 above)
Comment (m4s-b3n) [reply README.md:56]: @aatmmr thx

**Pull request merged**